apps/assisted_tagging/serializers.py

```diff
@@ -11,33 +10,35 @@
     PredictionTagAnalysisFrameworkWidgetMapping,
     WrongPredictionReview,
 )
-from .tasks import trigger_request_for_draft_entry_task
+from lead.models import Lead
+from .tasks import (
+    trigger_request_for_draft_entry_task,
+    trigger_request_for_auto_draft_entry_task
+)
 
 
 # ---------- Graphql ---------------------------
-class DraftEntryGqlSerializer(ProjectPropertySerializerMixin, UserResourceCreatedMixin, serializers.ModelSerializer):
-    class Meta:
-        model = DraftEntry
-        fields = (
-            'lead',
-            'excerpt',
-        )
-
+class DraftEntryBaseSerializer(serializers.Serializer):
     def validate_lead(self, lead):
         if lead.project != self.project:
             raise serializers.ValidationError('Only lead from current project are allowed.')
         af = lead.project.analysis_framework
         if af is None or not af.assisted_tagging_enabled:
             raise serializers.ValidationError('Assisted tagging is disabled for the Framework used by this project.')
-        return lead
-
-    def validate(self, data):
-        if self.instance and self.instance.created_by != self.context['request'].user:
-            raise serializers.ValidationError('Only reviewer can edit this review')
-        data['project'] = self.project
         if self.project.is_private:
             raise serializers.ValidationError('Assisted tagging is not available for private projects.')
-        return data
+        return lead
```

Comment:
We need to validate confidentiality permission here as well.

In lead/models.py
```python
from deep.permissions import ProjectPermissions as PP

...
class Lead:

    ...

    def has_permission(self, allowed_permissions):
        if (
            self.confidentiality != self.Confidentiality.UNPROTECTED and
            PP.Permission.VIEW_ALL_LEAD in allowed_permissions
        ):
            return True
        return PP.Permission.VIEW_ONLY_UNPROTECTED_LEAD in allowed_permissions
```

Then here,
```suggestion
        allowed_permission = self.context['request'].project_permissions
        if not lead.has_permission(allowed_permission):
            raise serializers.ValidationError("You don't have acccess to this lead")
        return lead
```

Reply:
Let's update this to throw error on RESTRICTED, CONFIDENTIAL without checking user authorizations.